### ISSUE TITLE
NPE in 9.0.0

### ISSUE THREAD

**Issue by amir-khassaia** [nebula-plugins/gradle-lint-plugin#164]
Updating to 9.x from 8.3.x results in a NullPointerException on lint.

Lint rules configured:
gradleLint.rules = ['duplicate-dependency-class', 'overridden-dependency-version']

```
09:12:03 Caused by: java.lang.NullPointerException: Cannot get property 'value' on null object
09:12:03 	at com.netflix.nebula.lint.rule.BuildFiles.original(BuildFiles.groovy:39)
09:12:03 	at com.netflix.nebula.lint.rule.BuildFiles$original.call(Unknown Source)
09:12:03 	at com.netflix.nebula.lint.GradleViolation.replaceWith(GradleViolation.groovy:114)
09:12:03 	at com.netflix.nebula.lint.rule.dependency.OverriddenDependencyVersionRule.visitGradleDependency(OverriddenDependencyVersionRule.groovy:42)
09:12:03 	at com.netflix.nebula.lint.rule.GradleLintRule.this$dist$invoke$2(GradleLintRule.groovy)
```

**Comment by chali** [comment]
Hi @amir-khassaia thank you for the bug report. Would you have an example project which could replicate it?

**Comment by amir-khassaia** [comment]
Unfortunately I dont have a replication in a simple project handy...

**Comment by chali** [comment]
I've looked at `overridden-dependency-version` rule which is failing and I haven't found anything obvious. I created this PR #167 with an improved error message which we can later try and get more details. A bit context of your error. We take all build files contributing to a project concatenate them to one file and then run all rules seeing all files together. Then when a violation is found we map concatenated line to an original file. In your case concatenated line doesn't point to a valid range. Do you use any `apply from: file.gradle` in your project? How looks like your dependencies blocks?

**Comment by chali** [comment]
Ok, the PR was merged and it is released as part of 9.1.0. Please let me know if you could get more details from improved exception message.

**Comment by amir-khassaia** [comment]
Hi, yes `apply from: ` is being used multiple times.

Stay tuned I will retest with 9.1.x. Will it be available on maven repos soon ?

**Comment by amir-khassaia** [comment]
The failure from 9.1 includes confirmation that some sort of mismatch with concatenated output and src lines is indeed going on:

```
00:01:37.541 Error from [com.netflix.nebula.lint.rule.GradleLintRule$1] processing source file [null]
00:01:37.556 
00:01:37.557 FAILURE: Build failed with an exception.
00:01:37.557 
00:01:37.557 * What went wrong:
00:01:37.558 Asked line in concatenated file was: 810 but it wasn't found. Original project files were concatenated to following ranges:
00:01:37.559 Lines 1 - 1 are C:\jenkins\workspace\papercut-clie---cf4be725\build.gradle
00:01:37.559 Lines 2 - 2 are C:\jenkins\workspace\papercut-clie---cf4be725\dependencies.gradle
00:01:37.560 Lines 3 - 3 are C:\jenkins\workspace\papercut-clie---cf4be725\build-translations.gradle
00:01:37.561 Lines 4 - 4 are C:\jenkins\workspace\papercut-clie---cf4be725\build-packaging.gradle
00:01:37.561 
00:01:37.562 * Try:
00:01:37.588 Run with --debug option to get more log output. Run with --scan to get full insights.
00:01:37.588 
00:01:37.588 * Exception is:
00:01:37.589 java.lang.IllegalArgumentException: Asked line in concatenated file was: 810 but it wasn't found. Original project files were concatenated to following ranges:
00:01:37.589 Lines 1 - 1 are C:\jenkins\workspace\papercut-clie---cf4be725\build.gradle
00:01:37.590 Lines 2 - 2 are C:\jenkins\workspace\papercut-clie---cf4be725\dependencies.gradle
00:01:37.591 Lines 3 - 3 are C:\jenkins\workspace\papercut-clie---cf4be725\build-translations.gradle
00:01:37.591 Lines 4 - 4 are C:\jenkins\workspace\papercut-clie---cf4be725\build-packaging.gradle
00:01:37.593 	at com.netflix.nebula.lint.rule.BuildFiles.original(BuildFiles.groovy:43)
00:01:37.593 	at com.netflix.nebula.lint.rule.BuildFiles$original.call(Unknown Source)
00:01:37.594 	at com.netflix.nebula.lint.GradleViolation.replaceWith(GradleViolation.groovy:114)
00:01:37.594 	at com.netflix.nebula.lint.rule.dependency.OverriddenDependencyVersionRule.visitGradleDependency(OverriddenDependencyVersionRule.groovy:42)
00:01:37.595 	at com.netflix.nebula.lint.rule.GradleLintRule.this$dist$invoke$2(GradleLintRule.groovy)
00:01:37.596 	at com.netflix.nebula.lint.rule.GradleLintRule$1.methodMissing(GradleLintRule.groovy)
00:01:37.596 	at com.netflix.nebula.lint.rule.GradleLintRule$1.this$dist$invoke$3(GradleLintRule.groovy)
00:01:37.597 	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.methodMissing(GradleLintRule.groovy)
00:01:37.598 	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.visitMethodCallInDependencies(GradleLintRule.groovy:435)
00:01:37.599 	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.visitMethodCallExpression(GradleLintRule.groovy:247)
00:01:37.601 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor$_visitMethodCallExpression_closure31.doCall(CompositeGroovyAstVisitor.groovy:284)
00:01:37.601 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitMethodCallExpression(CompositeGroovyAstVisitor.groovy:284)
00:01:37.601 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitExpressionStatement(CompositeGroovyAstVisitor.groovy)
00:01:37.602 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitExpressionStatement(CompositeGroovyAstVisitor.groovy:225)
00:01:37.602 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitBlockStatement(CompositeGroovyAstVisitor.groovy)
00:01:37.604 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitBlockStatement(CompositeGroovyAstVisitor.groovy:189)
00:01:37.604 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$2$visitClosureExpression(CompositeGroovyAstVisitor.groovy)
00:01:37.605 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitClosureExpression(CompositeGroovyAstVisitor.groovy:347)
00:01:37.606 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$2$visitListOfExpressions(CompositeGroovyAstVisitor.groovy)
00:01:37.606 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitListOfExpressions(CompositeGroovyAstVisitor.groovy:473)
00:01:37.608 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$2$visitTupleExpression(CompositeGroovyAstVisitor.groovy)
00:01:37.608 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitTupleExpression(CompositeGroovyAstVisitor.groovy:353)
00:01:37.609 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$2$visitArgumentlistExpression(CompositeGroovyAstVisitor.groovy)
00:01:37.609 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitArgumentlistExpression(CompositeGroovyAstVisitor.groovy:479)
00:01:37.610 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$2$visitMethodCallExpression(CompositeGroovyAstVisitor.groovy)
00:01:37.611 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitMethodCallExpression(CompositeGroovyAstVisitor.groovy:286)
00:01:37.612 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitExpressionStatement(CompositeGroovyAstVisitor.groovy)
00:01:37.612 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitExpressionStatement(CompositeGroovyAstVisitor.groovy:225)
00:01:37.613 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitBlockStatement(CompositeGroovyAstVisitor.groovy)
00:01:37.614 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitBlockStatement(CompositeGroovyAstVisitor.groovy:189)
00:01:37.614 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitClassCodeContainer(CompositeGroovyAstVisitor.groovy)
00:01:37.615 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitClassCodeContainer(CompositeGroovyAstVisitor.groovy:136)
00:01:37.616 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitConstructorOrMethod(CompositeGroovyAstVisitor.groovy)
00:01:37.617 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitConstructorOrMethod(CompositeGroovyAstVisitor.groovy:147)
00:01:37.617 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitMethod(CompositeGroovyAstVisitor.groovy)
00:01:37.618 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitMethod(CompositeGroovyAstVisitor.groovy:159)
00:01:37.619 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.super$3$visitClass(CompositeGroovyAstVisitor.groovy)
00:01:37.619 	at com.netflix.nebula.lint.rule.CompositeGroovyAstVisitor.visitClass(CompositeGroovyAstVisitor.groovy:105)
00:01:37.620 	at com.netflix.nebula.lint.org.codenarc.rule.AbstractAstVisitorRule.applyTo(AbstractAstVisitorRule.java:97)
00:01:37.621 	at com.netflix.nebula.lint.org.codenarc.rule.AbstractRule.applyTo(AbstractRule.java:142)
00:01:37.621 	at com.netflix.nebula.lint.org.codenarc.rule.Rule$applyTo$0.call(Unknown Source)
00:01:37.622 	at com.netflix.nebula.lint.rule.GradleLintRule.applyTo(GradleLintRule.groovy:184)
00:01:37.623 	at com.netflix.nebula.lint.org.codenarc.rule.Rule$applyTo$0.call(Unknown Source)
00:01:37.624 	at com.netflix.nebula.lint.org.codenarc.rule.Rule$applyTo$0.call(Unknown Source)
00:01:37.626 	at com.netflix.nebula.lint.org.codenarc.analyzer.AbstractSourceAnalyzer.collectViolations(AbstractSourceAnalyzer.groovy:40)
00:01:37.626 	at com.netflix.nebula.lint.plugin.LintService$ReportableAnalyzer.analyze(LintService.groovy:50)
00:01:37.626 	at com.netflix.nebula.lint.plugin.LintService$ReportableAnalyzer$analyze.call(Unknown Source)
00:01:37.627 	at com.netflix.nebula.lint.plugin.LintService$_lint_closure5.doCall(LintService.groovy:122)
00:01:37.628 	at com.netflix.nebula.lint.plugin.LintService.lint(LintService.groovy:111)
00:01:37.629 	at com.netflix.nebula.lint.plugin.LintService$lint.call(Unknown Source)
00:01:37.630 	at com.netflix.nebula.lint.plugin.LintGradleTask.lint(LintGradleTask.groovy:37)
00:01:37.630 	at com.netflix.nebula.lint.plugin.LintGradleTask$lint.call(Unknown Source)
00:01:37.631 	at com.netflix.nebula.lint.plugin.GradleLintPlugin$1.buildFinished(GradleLintPlugin.groovy:64)
00:01:37.631 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
00:01:37.632 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
00:01:37.632 	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:370)
00:01:37.633 	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:352)
00:01:37.634 	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:58)
00:01:37.635 	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast$ListenerDispatch.dispatch(DefaultListenerManager.java:340)
00:01:37.635 	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast$ListenerDispatch.dispatch(DefaultListenerManager.java:327)
00:01:37.636 	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
00:01:37.637 	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:230)
00:01:37.638 	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:149)
00:01:37.638 	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:58)
00:01:37.639 	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:324)
00:01:37.640 	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:234)
00:01:37.640 	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:140)
00:01:37.641 	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:37)
00:01:37.641 	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
00:01:37.642 	at com.sun.proxy.$Proxy17.buildFinished(Unknown Source)
00:01:37.643 	at org.gradle.initialization.DefaultGradleLauncher.finishBuild(DefaultGradleLauncher.java:154)
00:01:37.643 	at org.gradle.initialization.DefaultGradleLauncher.finishBuild(DefaultGradleLauncher.java:122)
00:01:37.644 	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:141)
00:01:37.645 	at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:115)
00:01:37.646 	at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:78)
00:01:37.646 	at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:75)
00:01:37.647 	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:152)
00:01:37.647 	at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:100)
00:01:37.648 	at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:75)
00:01:37.649 	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
00:01:37.650 	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
00:01:37.651 	at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
00:01:37.652 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$1.run(RunAsBuildOperationBuildActionRunner.java:43)
00:01:37.652 	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
00:01:37.653 	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
00:01:37.654 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
00:01:37.655 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
00:01:37.655 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:40)
00:01:37.656 	at org.gradle.tooling.internal.provider.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:51)
00:01:37.657 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:49)
00:01:37.657 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:32)
00:01:37.658 	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:39)
00:01:37.659 	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:25)
00:01:37.660 	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:80)
00:01:37.660 	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:53)
00:01:37.661 	at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:57)
00:01:37.662 	at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:32)
00:01:37.662 	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:36)
00:01:37.663 	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:25)
00:01:37.663 	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
00:01:37.664 	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
00:01:37.665 	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:64)
00:01:37.666 	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:29)
00:01:37.666 	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:59)
00:01:37.667 	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:44)
00:01:37.679 	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:45)
00:01:37.680 	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:30)
00:01:37.680 	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
00:01:37.680 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
00:01:37.681 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.682 	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
00:01:37.682 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.683 	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
00:01:37.684 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.685 	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
00:01:37.685 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.685 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
00:01:37.686 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
00:01:37.687 	at org.gradle.util.Swapper.swap(Swapper.java:38)
00:01:37.688 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
00:01:37.688 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.689 	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
00:01:37.690 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.691 	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:62)
00:01:37.691 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
00:01:37.692 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.693 	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:82)
00:01:37.694 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
00:01:37.694 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
00:01:37.695 	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
00:01:37.695 	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:295)
00:01:37.696 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
00:01:37.697 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
00:01:37.697 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
```

**Comment by chali** [comment]
Thanks, @amir-khassaia this was super helpful. We have a bug in counting lines of files on Windows. I will look into it.

**Comment by chali** [comment]
I released 9.3.1. The issue should be fixed. @amir-khassaia can you give it a try?

**Comment by amir-khassaia** [comment]
Looks like this issue is addressed
